Automattic/jetpack
# Use Modules class in My Jetpack

#### Changes proposed in this Pull Request:

For Hybrid products use the new `Modules` class to activate modules and check if product is active

This allows hybrid products such as Search and Social to better integrate with My Jetpack, as they also use the Modules abstraction

#### Jetpack product discussion


p9dueE-55u-p2#comment-7978

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Spin up a new JN site with Jetpack Beta
* Activate Boost and Search on this branch
* Deactivate and delete Jetpack
* Connect Boost
* Go to My Jetpack and connect your user (we need to do this while we don't fix the purchase flow)
* Back to My Jetpack, click "Purchase" on the Search card and follow the flow. Buy Search
* You should go back to the Search dashboard with all enabled
* Go to My Jetpack and confirm the card shows Search as active with a "Manage" button
* Click Manage and confirm you go to the Search plugin page
* Deactivate Search (the toggle on the Search dashboard, not the plugin)
* Go back to My Jetpack and confirm the product appears as not active, with an "Activate" button
* Click Activate
* Go to Search page and confirm Search and Instant Search are re-enabled
